forge-gui/res/cardsfolder/upcoming/heros_heirloom.txt

```diff
@@ -0,0 +1,9 @@
+Name:Hero's Heirloom
+ManaCost:2
+Types:Artifact Equipment
+K:Equip:2
+S:Mode$ Continuous | Affected$ Creature.EquippedBy | AddPower$ 2 | AddToughness$ 1 | Description$ Equipped creature gets +2/+1.
+S:Mode$ Continuous | Affected$ Card.EquippedBy+Legendary | AddKeyword$ Trample & Haste | Description$ As long as equipped creature is legendary, it has trample and haste.
+AI:RemoveDeck:Random
```

Comment:
I don't think this is necessary. It's a pretty good card for {2} even if equipped on a nonlegendary. A hint is enough to make it more appropriately used in random decks, but if you want, you can promote it to DeckNeeds so that the AI only puts one in deck if it has at least one legendary.